core/common/src/main/java/alluxio/network/TieredIdentityFactory.java

```diff
@@ -98,37 +108,50 @@ private static TieredIdentity fromScript() {
     if (!Configuration.containsKey(PropertyKey.LOCALITY_SCRIPT)) {
       return null;
     }
-    String script = Configuration.get(PropertyKey.LOCALITY_SCRIPT);
+    Path script = Paths.get(Configuration.get(PropertyKey.LOCALITY_SCRIPT));
+    if (!Files.exists(script)) {
+      return null;
+    }
     String identityString;
     try {
-      identityString = ShellUtils.execCommand(script);
+      identityString = ShellUtils.execCommand(script.toString());
     } catch (IOException e) {
       throw new RuntimeException(
           String.format("Failed to run script %s: %s", script, e.toString()), e);
     }
-    return fromString(identityString);
+    try {
+      return fromString(identityString);
+    } catch (IOException e) {
+      throw new RuntimeException(
+          String.format("Failed to parse output of running %s: %s", script, e.getMessage()), e);
+    }
   }
 
   /**
    * @param identityString tiered identity string to parse
    * @return the parsed tiered identity
    */
-  public static TieredIdentity fromString(String identityString) {
+  public static TieredIdentity fromString(String identityString) throws IOException {
+    Set<String> allTiers = Sets.newHashSet(Configuration.getList(PropertyKey.LOCALITY_ORDER, ","));
     Map<String, String> tiers = new HashMap<>();
     for (String tier : identityString.split(",")) {
       String[] parts = tier.split("=");
       if (parts.length != 2) {
-        throw new RuntimeException(String
+        throw new IOException(String
```

Comment:
Add exception strings to `ExceptionMessage`?

Reply:
I prefer it this way so that the formatting placeholders are defined in the same place as their replacements. It also serves as documentation for the code, and makes it easier to get to this file directly by grepping.